packages/message-encryption/src/ecies.ts

```diff
@@ -193,7 +202,19 @@ class Decoder extends DecoderV0 implements IDecoder<DecodedMessage> {
 export function createDecoder(
   contentTopic: string,
   privateKey: Uint8Array,
-  pubsubTopic: PubSubTopic = DefaultPubSubTopic
+  pubsubTopic: SingleTopicShardInfo | PubsubTopic = DefaultPubSubTopic
 ): Decoder {
-  return new Decoder(pubsubTopic, contentTopic, privateKey);
+  if (typeof pubsubTopic === "string" && pubsubTopic !== DefaultPubSubTopic) {
```

Comment:
Since anything but `SingleTopicShardInfo` and `DefaultPubsubTopic` are allowed then we can simply do:
```ts
export function createDecoder(
  contentTopic: string,
  privateKey: Uint8Array,
  pubsubTopic?: SingleTopicShardInfo
): Decoder {
```

and then 


```ts
return new Decoder(
    pubsubTopic?.index ?
      : singleTopicShardInfoToPubSubTopic(pubsubTopic)
      : DefaultPubsubTopic,
    contentTopic,
    privateKey
  );
```

Reply:
of course it still can be thrown, but inside of `singleTopicShardInfoToPubSubTopic` since it will fail to parse passed pubsub